src/simplicial-hott/06-2cat-of-segal-types.rzk.md

```diff
@@ -80,3 +81,148 @@ Functions between types automatically preserve identity arrows.
+
+```rzk
+#def nat-trans-components
+	(A : U)
+	(B : A -> U)
+	(f g : (x : A) -> (B x))
+	: U
+	:= (x : A) -> hom (B x) (f x) (g x)
+```
+
+```rzk
+#def ev-components-nat-trans
+	(A : U)
+	(B : A -> U)
+	(f g : (x : A) -> (B x))
+	(η : nat-trans A B f g)
+	: nat-trans-components A B f g
+	:= \ x t -> η t x
+
+#def nat-trans-nat-trans-components
+	(A : U)
+	(B : A -> U)
+	(f g : (x : A) -> (B x))
+	(η : nat-trans-components A B f g)
+	: nat-trans A B f g
+	:= \ t x -> η x t
+```
+
+### Natural transformation extensionality
+
+```rzk
+-- [RS17, Proposition 6.3]
+#def is-equiv-ev-components-nat-trans
+	(A : U)
+	(B : A -> U)
+	(f g : (x : A) -> (B x))
+	: isEquiv
+      (nat-trans A B f g)
+      (nat-trans-components A B f g)
+      (ev-components-nat-trans A B f g)
+	:=
+		( ( \ η t x -> η x t , \ _ -> refl) ,
+		  ( \ η t x -> η x t , \ _ -> refl))
+
+#def equiv-components-nat-trans
+	(A : U)
+	(B : A -> U)
+	(f g : (x : A) -> (B x))
+	: Eq
+      (nat-trans A B f g)
+      (nat-trans-components A B f g)
+	:=
+		( ev-components-nat-trans A B f g ,
+      is-equiv-ev-components-nat-trans A B f g)
+```
+
+### Horizontal composition
+
+Horizontal composition of natural transformations makes sense over any type.
+Note that contrary to what is written in [RS17] we do not need `C` to be Segal.
+
+```rzk
+#def horizontal-comp-nat-trans
+	(A B C : U)
+	(f g : A -> B)
+	(f' g' : B -> C)
+	(η : nat-trans A (\ _ -> B) f g)
+	(η' : nat-trans B (\ _ -> C) f' g')
+	: nat-trans A (\ _ -> C) (\ x -> f' (f x)) (\ x -> g' (g x))
+	:= \ t x -> η' t (η t x)
+
+#def horizontal-comp-nat-trans-components
+	(A B C : U)
+	(f g : A -> B)
+	(f' g' : B -> C)
+	(η : nat-trans-components A (\ _ -> B) f g)
+	(η' : nat-trans-components B (\ _ -> C) f' g')
+	: nat-trans-components A (\ _ -> C) (\ x -> f' (f x)) (\ x -> g' (g x))
+	:= \ x t -> η' (η x t) t
+```
+
+### Vertical composition
+
+We can define vertical composition for natural transformations in families of
+Segal types.
+
+```rzk
+#def vertical-comp-nat-trans-components
+	(A : U)
+	(B : A -> U)
+	(BisSegal : (x : A) -> isSegal (B x))
+	(f g h : (x : A) -> (B x))
+	(η : nat-trans-components A B f g)
+	(η' : nat-trans-components A B g h)
+	: nat-trans-components A B f h
+	:= \ x -> Segal-comp (B x) (BisSegal x) (f x) (g x) (h x) (η x) (η' x)
+
+#def vertical-comp-nat-trans
+	(A : U)
+	(B : A -> U)
+	(BisSegal : (x : A) -> isSegal (B x))
+	(f g h : (x : A) -> (B x))
+	(η : nat-trans A B f g)
+	(η' : nat-trans A B g h)
+	: nat-trans A B f h
+	:=
+    \ t x ->
+    vertical-comp-nat-trans-components A B BisSegal f g h
+      ( \ x t -> η t x)
+      ( \ x t -> η' t x)
+      ( x)
+      ( t)
+```
+
+The identity natural transformation is identity arrows on components
+
+```rzk
+-- [RS17, Proposition 6.5(ii)]
+#def id-arr-components-id-nat-trans
+	(A : U)
+	(B : A -> U)
+	(f : (x : A) -> (B x))
+  (x : A)
+  : (\{t : 2 | Δ¹ t} -> (id-arr ((x : A) -> (B x)) f) t x) = (id-arr (B x) (f x))
```

Comment:
I wonder if this can be simplified (type annotation removed):
```suggestion
  : (\t -> id-arr ((z : A) -> B z) f t x) = id-arr (B x) (f x)
```
or
```suggestion
  : (\t -> id-arr ((z : A) -> B z) f t x) =_{Δ¹ -> B x} id-arr (B x) (f x)
```
Also, I think here name shadowing occurs for `x`, so I renamed one to `z`.

Reply:
Thanks! I went with
```rzk
  : (\ t -> id-arr ((x : A) -> B x) f t a) =_{Δ¹ -> B a} id-arr (B a) (f a)
```
renaming the outer `x` to `a` :)

Reply:
The first one did not work as the type checker couldn't infer the shape